### DISCUSSION TITLE
No emulator found that matches 'emulator'.

### DISCUSSION THREAD

**Discussion by olheimer** [codemagic-ci-cd/codemagic-docs#1353] [Q&A] [answered]
Hi,
I'm new to code magic. 
I'm trying to get Flutter integration tests running both ion iOS and Android emulator. iOS seems no problem.
But Android does not work.

When looking in the "Workflow Editor" I can't select Android Emulator (only iOS or chrome).

So I thought starting it in my codemagic.yaml and let the tests run (my codemagic.yaml):
android-workflow:
    name: Android Workflow
    max_build_duration: 60
    environment:
      flutter: default
      xcode: latest
      cocoapods: default
    scripts:
      - flutter emulators --launch emulator
      - flutter packages pub get
      - flutter test integration_test
      - flutter test
    artifacts:
      - flutter_drive.log
    publishing:
      email:
        recipients:
          - abc@zde.de

But in the build run I do get the following error on " flutter emulators --launch emulator":
No emulator found that matches 'emulator'.

And running the integration tests with "flutter test integration_test" I do get the following error:
unning "flutter pub get" in clone...                              594ms
More than one device connected; please specify a device with the '-d <deviceId>' flag, or use '-d all' to act on all devices.

macOS (desktop) • macos  • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm
Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119
No devices are connected. Ensure that `flutter doctor` shows at least one connected device

As said above iOS works. Is there any reason why I can't use Android emulator in a personal account without billing? I'm using the only available machine: Machine:
Mac mini M1

**Comment by himesh-cm** [comment]
Hi @olheimer ,
you need to await for the emulator succesfully to launch before running the tests.
```
flutter emulators --launch emulator & adb wait-for-device
```

**Comment by olheimer** [comment, marked as answer]
Hi @himesh-cm,

unfortunately this does not help. When launching the emulator I do get the following error:
* daemon not running; starting now at tcp:5037
* daemon started successfully
No emulator found that matches 'emulator'.

**Reply by himesh-cm** [reply]
Can you please share the build id?

**Reply by olheimer** [reply]
63d782ab9d555029bc186612

**Reply by himesh-cm** [reply]
Hi @olheimer, Android Emulators are not available with M1 Machines. You can use `mac_pro` or `linux` instances to run tests on android.
It’s a limitation by Apple Silicon that the Apple Virtualization API doesn’t support nested virtualization, which is required for emulators.
